.github/actions/simeng_benchmarks/action.yml

```diff
@@ -2,48 +2,45 @@ name: simeng-benchmarks
 description: runs simeng benchmarks
 
 inputs:
-  BENCHMARK_BRANCH:
-    description: name of the branch on the benchmark repo you want to checkout 
+  RUN_DIR:
+    description: directory from which the benchmark binary should be run
+    required: false
+  DATA_PATH:
+    description: location of the data file(s) for this benchmarks
+    required: false
+  BIN_PATH:
+    description: path to the binary for the benchmark
     required: true
-  OS:
-    description: docker image name
-    required: true
-  PAT:
-    description: personal access token required to checkout private benchmark repo
+  PASS_STRING:
+    description: string that is searched for in the benchmark's output to confirm whether or not it has passed
     required: true
 
 runs:
   using: 'composite'
   steps:
-
-    ##################################
-    # clones repo with different versions of the checkout action, as container has different node versions and some compilers run on older OSs that have differening node versions
-    
-
-    - if: ${{ contains(fromJson('["ubuntu:18.04"]'), matrix.OS) }}
-      name: checkout v3
-      uses: actions/checkout@v3
-      with:
-        repository: UoB-HPC/simeng-benchmarks
-        token: ${{ inputs.PAT }}
-        ref: makefile-build-system
-        path: simeng-benchmarks
-
-    - if: ${{ !contains(fromJson('["ubuntu:18.04"]'), matrix.OS) }}
-      name: checking out benchmark repository
-      uses: actions/checkout@v4
-      with:
-        repository: UoB-HPC/simeng-benchmarks
-        ref: makefile-build-system
-        token: ${{ inputs.PAT }}
-        path: simeng-benchmarks
-
-    - name: run benchmark
+    - name: Run Benchmark
       shell: bash
       run: |
-        cd ./simeng-benchmarks/Tools/simulation/validation
-        SIMENG_BENCHMARKS_SRC_DIR=$GITHUB_WORKSPACE/simeng-benchmarks python3 validate_simulation.py all_a64fx.json $GITHUB_WORKSPACE/configs/a64fx.yaml
-
+        if [ ${{ inputs.DATA_PATH }} ]
+        then
+          simeng "$GITHUB_WORKSPACE/configs/a64fx.yaml" "${{ inputs.BIN_PATH }}" -n 64 -i 1 --deck "${{ inputs.DATA_PATH }}"  > $GITHUB_WORKSPACE/simeng.tmp
+        elif [ ${{ inputs.RUN_DIR }} ]
+        then
+          cd ${{ inputs.RUN_DIR }}
+          simeng "$GITHUB_WORKSPACE/configs/a64fx.yaml" "${{ inputs.BIN_PATH }}" -n 64 -i 1  > $GITHUB_WORKSPACE/simeng.tmp
+        else
+          simeng "$GITHUB_WORKSPACE/configs/a64fx.yaml" "${{ inputs.BIN_PATH }}" -n 64 -i 1  > $GITHUB_WORKSPACE/simeng.tmp
+        fi
```

Comment:
I think in all cases, all benchmarks are being executed with hardcoded miniBUDE parameters. AFAIK, for our specific workloads atm this shouldn't cause issues, but it isn't future proof. As per above, if the whole string "-n 64 -i 1 /path/to/bm1" were passed into this action then this part would simplify a lot and should prevent issues. Tell me if there is anything that actually prevents you from doing that